src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,986 @@
+                std::move(s.modData()),
+                node.getNodeHash().as_uint256())};
+            if (!BEAST_EXPECT(nSrc))
+                return false;
+
+            auto nDst =
+                db->fetch(node.getNodeHash().as_uint256(), l->info().seq);
+            if (!BEAST_EXPECT(nDst))
+                return false;
+
+            BEAST_EXPECT(isSame(nSrc, nDst));
+
+            return true;
+        };
+        if (l->info().txHash.isNonZero())
+            l->txMap().snapShot(false)->visitNodes(fcompTx);
+    }
+
+    std::string
+    bitmask2Rangeset(std::uint64_t bitmask)
+    {
+        std::string set;
+        if (!bitmask)
+            return set;
+        bool empty = true;
+
+        for (std::uint32_t i = 0; i < 64 && bitmask; i++)
+        {
+            if (bitmask & (1ll << i))
+            {
+                if (!empty)
+                    set += ",";
+                set += std::to_string(i);
+                empty = false;
+            }
+        }
+
+        RangeSet<std::uint32_t> rs;
+        from_string(rs, set);
+        return to_string(rs);
+    }
+
+    std::unique_ptr<Config>
+    testConfig(
+        std::string const& testName,
+        std::string const& backendType,
+        std::string const& shardDir,
+        std::string const& nodeDir = std::string())
+    {
+        using namespace test::jtx;
+
+        if (testName != "")
+        {
+            std::string caseName =
+                "DatabaseShard " + testName + " with backend " + backendType;
+            testcase(caseName);
+        }
+
+        return envconfig([&](std::unique_ptr<Config> cfg) {
+            cfg->overwrite(ConfigSection::shardDatabase(), "type", backendType);
+            cfg->overwrite(ConfigSection::shardDatabase(), "path", shardDir);
+            cfg->overwrite(
+                ConfigSection::shardDatabase(),
+                "max_size_gb",
+                std::to_string(maxSizeGb));
+            cfg->overwrite(
+                ConfigSection::shardDatabase(),
+                "ledgers_per_shard",
+                std::to_string(ledgersPerShard));
+            cfg->overwrite(
+                ConfigSection::shardDatabase(),
+                "earliest_seq",
+                std::to_string(earliestSeq));
+            cfg->overwrite(ConfigSection::nodeDatabase(), "type", backendType);
+            cfg->overwrite(
+                ConfigSection::nodeDatabase(),
+                "max_size_gb",
+                std::to_string(maxSizeGb));
+            cfg->overwrite(
+                ConfigSection::nodeDatabase(),
+                "earliest_seq",
+                std::to_string(earliestSeq));
+            if (nodeDir.empty())
+                cfg->overwrite(
+                    ConfigSection::nodeDatabase(), "path", defNodeDir.path());
+            else
+                cfg->overwrite(ConfigSection::nodeDatabase(), "path", nodeDir);
+            return cfg;
+        });
+    }
+
+    int
```

Comment:
I prefer `optional` to returning `-1`. Fine for this test code if you want to keep it tho.

Reply:
Done